eclipse-plugin/service/cc.codechecker.service.api/src/main/java/cc/codechecker/api/runtime/CodecheckerServerThread.java

```diff
@@ -11,7 +11,8 @@
 
 public class CodecheckerServerThread {
 
-    public final int serverPort = (new Random()).nextInt(10000) + 15000;
+    public final int serverPort = (new Random()).nextInt(10000) + 5000 + Math.min(
```

Comment:
This can still return the same number if you are unlucky. Java's random seed is millisecond based, you just lowered the chance of every random happeining in the same millisec - it can still happen on a fast PC. Instead of adding a bunch of more random generators, extract the random generator creation from this into a fixed scope - inject it as a constructor parameter, or make it static, so it is only constructed once.